torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2691,16 +2688,18 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::collective(
   work->outputs_ =
       std::make_shared<std::vector<at::Tensor>>(std::move(outputs));
 
-  if (avoidRecordStreams) {
+  if (avoidRecordStreams && inputs.size() == 0) {
     work->stashed_for_allocator_safety_ =
         std::make_shared<std::vector<at::Tensor>>();
-    work->stashed_for_allocator_safety_->push_back(input);
+    work->stashed_for_allocator_safety_->push_back(inputs[0]);
   }
```

Comment:
Maybe check if we need to push the entire inputs into `stashed_for_allocator_safety_`

Reply:
For alltoall, we do this in preprocess func.. lol